cves/CVE-2016-4433.yml

```diff
@@ -205,4 +224,13 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    Looking at the fix I notice they're disallowing arithmetic expressions, so
+    perhaps plus symbols were being expanded to spaces later for example,
+    causing issues. Looking into the test added at the fix commit it expects a
+    method call of "#booScope=@myclass@DEFAULT_SCOPE,#bootScope.init()" to fail.
+
+    Given that this syntax appears to allow arbitrary method calls from a remote
+    source it seems like the use of it is a design flaw, as Apache will be
+    forever plugging holes in this system. Instead the app should be required to
+    define its input paths, instead of letting the client do whatever it wants.
```

Comment:
Is there a way that this particular mistake could be avoided in similar projects in the future?